### ISSUE TITLE
Add drag and drop feature for local files

### ISSUE THREAD

**Issue by poush** [poush/H2#12]
It should support only certain file types. Allowing people to drag and drop it to the app to view it in picture in picture mode.

**Comment by poush** [comment]
This also adds the question of adding the dock icon back for macOS or not.

**Comment by rouzbehhz** [comment]
Just submitted a PR. 

![demo-h2](https://user-images.githubusercontent.com/40371763/46556729-87560f00-c8b5-11e8-86d3-ea3faa8caf7a.gif)

I have a bunch of cool ideas to add to this project.

**Comment by poush** [comment]
@rouzbehhz no hurries, just wanted if you are currently working on re-submitting that PR because I have one idea on this. Actually, we can rather directly connecting it with a specific feature, connect it with the providers module. Thus, if anyone drags any file which will call providers with the local URI. Providers on matching it with the respective provider of that URI will fire proper event in renderer process to process it. We can definitely work on this in new PR and issue.

**Comment by rouzbehhz** [comment]
@poush Yes, out cold last few days. Going to add this in sometime today or tomorrow, have a bunch of stuff from work I have to catch up on. Also added a nice app navigation and landing page I can include in the PR (Issue #24) if you want to wait on it. 

Cheers

**Comment by poush** [comment]
Thank @rouzbehhz. I hope to see further contributions to providers standardization :)
